src/services/expences.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+let expences = [];
+let expencesIdCount = 0;
+
+function resetExpences() {
+  expences = [];
+  expencesIdCount = 0;
+}
+
+function create(userData) {
+  expencesIdCount++;
+
+  const newExpence = {
+    id: expencesIdCount,
+    ...userData,
+  };
+
+  expences.push(newExpence);
+
+  return newExpence;
+}
+
+function filterAll({ userId, categories, from, to }) {
+  let filteredExpences = [...expences];
+
+  if (userId) {
+    filteredExpences = expences.filter((expence) => expence.userId === +userId);
```

Comment:
Replace 'expences' with 'filteredExpences' to correctly filter the results based on previous filters.